ReSwift/CoreTypes/ActionCreator.swift

```diff
@@ -10,9 +10,18 @@ import Foundation
 
 public protocol ActionCreator {
     
-    associatedtype State
+    associatedtype State: StateType
+    /**
+     Generates one or more new actions from the current state.  May be implemented asynchronously or synchronously
+     - parameter actionBlock This may be called more than once with different actions
+     */
+    func createActions(currentState state: State, actionBlock:(Action -> Void))
```

Comment:
@jonathanellis ActionCreator doesn't dispatch the actions, it just creates them.  
Some other object is responsible for dispatching them